juju_k8s_crashdump/main.py

```diff
@@ -0,0 +1,53 @@
+#!/bin/env/python3
+# Copyright 2025 Canonical Ltd.
+# See LICENSE file for licensing details.
+
+import argparse
+import os
+import tarfile
+from datetime import datetime
+from tempfile import TemporaryDirectory
+
+from .juju_cmd import JujuCmdClient
+from .k8s_cmd import KubectlCmdClient
+
+
+def create_parser():
+    parser = argparse.ArgumentParser(description="Collect logs for standard resources juju creates in kubernetes")
+    parser.add_argument("kubeconf", help="Path to a kubeconf with permissions to reach the resources juju creatres")
+    parser.add_argument("controller", help="Name of the controller to get logs for")
+    return parser
+
+
+def main():
+    parser = create_parser()
+    args = parser.parse_args()
+    output_tar = f"{os.getcwd()}/{datetime.now().strftime('%Y-%m-%d-%H.%M.%S')}.tar.gz"
+
+    juju_client = JujuCmdClient()
+    kubectl_client = KubectlCmdClient(args.kubeconf)
+    namespaces = [f"controller-{args.controller}"]
+    for model in juju_client.models(args.controller):
```

Comment:
I wouldn't be opposed to this being split out into a few functions, like `get_namespaces`, `save_resource`, `save_pod_logs`, `write_tar_file`. Fine if they're locally kept in this file, just think it'll make reading it a bit easier.